### PR TITLE
Ensure that the virtualenv dir is ignored from Git

### PR DESCRIPTION
**Pull request by SanketDG**
Fixes https://github.com/pypa/virtualenv/issues/1806

### Thanks for contributing, make sure you address all the checklists (for details on how see
[development documentation](https://virtualenv.pypa.io/en/latest/development.html#development))!

- [x] ran the linter to address style issues (``tox -e fix_lint``)
- [x] wrote descriptive pull request text
- [ ] ensured there are test(s) validating the fix (need help)
- [x] added news fragment in ``docs/changelog`` folder
- [] updated/extended the documentation

### CODE REVIEW

**Comment by SanketDG** [comment]
Question: Do we want this as a flag?

File: src/virtualenv/create/creator.py [hunk 1]
```diff
@@ -154,6 +154,15 @@ def run(self):
             safe_delete(self.dest)
         self.create()
         self.set_pyenv_cfg()
+        self._ensure_gitignore()
+
+    def _ensure_gitignore(self):
```
Comment (gaborbernat) [inline src/virtualenv/create/creator.py:159]: On second thought this needs to be within https://github.com/pypa/virtualenv/blob/master/src/virtualenv/create/via_global_ref/api.py#L89

File: src/virtualenv/create/creator.py [hunk 2]
```diff
@@ -154,6 +154,15 @@ def run(self):
...
+        """
+        Ignore the virtualenv directory from git.
+        """
+        self.dest.joinpath(".gitignore").write_text(
```
Comment (gaborbernat) [inline src/virtualenv/create/creator.py:163]: use the ``/`` instead of join path

Comment (gaborbernat) [reply src/virtualenv/create/creator.py:163]: the utf-8 encoding is not needed can leave it empty to use platform default;

the comment is not needed, please remove

File: docs/changelog/1806.feature.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Ensure that the virtualenv dir created is ignored by Git
```
Comment (gaborbernat) [inline docs/changelog/1806.feature.rst:1]: This does not follow the example structure, e.g. missing user

File: src/virtualenv/create/creator.py (hunk 1 above)
Comment (gaborbernat) [inline src/virtualenv/create/creator.py:159]: let's not make it git specific, let's cover all VCS covered here https://www.python.org/dev/peps/pep-0610/#registered-vcs

a better name is: ``generate_vcs_ignore``, no need to make it private

Comment (gaborbernat) [reply src/virtualenv/create/creator.py:159]: And yeah please add an option to disable this via ``--no-vcs-ingore`` 👍

**Comment by gaborbernat** [comment]
Superseded by https://github.com/pypa/virtualenv/pull/1825

**Pull request closed without merging**